udata/features/territories/commands.py

```diff
@@ -50,70 +53,77 @@ def migrate_zones_ids():
+        counter['datasets'] += 1
+        new_zones = []
+        for zone in dataset.spatial.zones:
+            if zone.id.startswith('fr/'):
+                counter['zones'] += 1
+                country, kind, zone_id = zone.id.split('/')
+                zone_id = zone_id.upper()  # Corsica 2a/b case.
+                if kind == 'town':
+                    counter['towns'] += 1
+                    new_zones.append(
+                        GeoZone
+                        .objects(code=zone_id, level='fr:commune')
+                        .valid_at(date.today())
+                        .first())
+                elif kind == 'county':
+                    counter['counties'] += 1
+                    new_zones.append(
+                        GeoZone
+                        .objects(code=zone_id, level='fr:departement')
+                        .valid_at(date.today())
+                        .first())
+                elif kind == 'region':
+                    counter['regions'] += 1
+                    # Only link to pre-2016 regions which kept the same id.
+                    new_zones.append(
+                        GeoZone
+                        .objects(code=zone_id, level='fr:region')
+                        .first())
                 else:
                     new_zones.append(zone)
-            dataset.update(
-                spatial=SpatialCoverage(zones=[z.id for z in new_zones if z]))
-    print('{} datasets and {} zones affected.'.format(
-        counter_datasets, counter_zones))
-    print('{} town zones, {} county zones and {} region zones updated.'.format(
-        counter_towns, counter_counties, counter_regions))
-    print('{} France zones, {} DROM zones, {} DROM-COM zones updated.'.format(
-        counter_france, counter_drom, counter_dromcom))
+            elif zone.id.startswith('country-subset/fr'):
+                counter['zones'] += 1
+                subset, country, kind = zone.id.split('/')
+                if kind == 'dom':
+                    counter['drom'] += 1
+                    new_zones.append(drom_zone)
+                elif kind == 'domtom':
+                    counter['dromcom'] += 1
+                    new_zones.append(dromcom_zone)
+            elif zone.id.startswith('country/'):
+                counter['zones'] += 1
+                counter['countries'] += 1
+                new_zones.append(zone.id.replace('/', ':'))
+            elif zone.id.startswith('country-group/'):
+                counter['zones'] += 1
+                counter['countrygroups'] += 1
+                new_zones.append(zone.id.replace('/', ':'))
+            # TODO: handle EPCIs
+            else:
+                new_zones.append(zone)
+        dataset.update(
+            spatial=SpatialCoverage(
+                granularity=dataset.spatial.granularity,
+                zones=[getattr(z, 'id', z) for z in new_zones if z]
+            )
+        )
+    log.info(Formatter().vformat('''Summary
```

Comment:
Yes, I just found it! I needed to use the default value (because the `**` doesn't works when the value is not defined)